From 668857a8eff24f564c6e148cb7de3da1e828a3dd Mon Sep 17 00:00:00 2001
From: Brandon McAnsh <brandonm@matricom.net>
Date: Wed, 16 Apr 2014 14:57:27 -0400
Subject: [PATCH 2/4] DVDFactoryCodec: Remove throttling of mpeg2 to libmpeg2
 for amlogic

* Thanks to @kszaq, am-mpeg2 can handle all mpeg2's however it can't handle the still frames (which is caught above it).
* Testing .VOB isn't setting hint.stills to true so will need to address this in separate patch

Signed-off-by: Brandon McAnsh <brandonm@matricom.net>
---
 xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp |   10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp
index d9d5b47..9e7ea0c 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp
@@ -204,15 +204,7 @@ CDVDVideoCodec* CDVDFactoryCodec::CreateVideoCodec(CDVDStreamInfo &hint, unsigne
   // amcodec can handle dvd playback.
   if (!hint.software && CSettings::Get().GetBool("videoplayer.useamcodec"))
   {
-    // but dvd's have weird still-frames in it, which is not fully supported in ffmpeg
-    if (hint.codec == AV_CODEC_ID_MPEG2VIDEO || hint.codec == AV_CODEC_ID_MPEG1VIDEO)
-    {
-      if( (pCodec = OpenCodec(new CDVDVideoCodecLibMpeg2(), hint, options)) ) return pCodec;
-    }
-    else
-    {
-      if ( (pCodec = OpenCodec(new CDVDVideoCodecAmlogic(), hint, options)) ) return pCodec;
-    }
+     if ( (pCodec = OpenCodec(new CDVDVideoCodecAmlogic(), hint, options)) ) return pCodec;
   }
 #endif
 
-- 
1.7.9.5

